src/coreclr/jit/indirectcalltransformer.cpp

```diff
@@ -1016,25 +1019,57 @@ class IndirectCallTransformer
         {
             // Compute likelihoods
             //
-            unsigned const thenLikelihood   = origCall->GetGDVCandidateInfo(checkIdx)->likelihood;
-            weight_t       thenLikelihoodWt = min(((weight_t)thenLikelihood) / 100.0, 100.0);
-            weight_t       elseLikelihoodWt = max(1.0 - thenLikelihoodWt, 0.0);
+            // If this is the first check the likelihood is just the candidate likelihood.
+            // If there are multiple checks things are a bit more complicated.
+            //
+            // Say we had three candidates with likelihoods 0.5, 0.3, and 0.1.
+            //
+            // The first one's likelihood is 0.5.
+            //
+            // The second one (given that we've already checked the first and failed)
+            // is (0.3) / (1.0 - 0.5) = 0.6.
+            //
+            // The third one is (0.1) / (1.0 - (0.5 + 0.2)) = 0.5
+            //
+            // So to figure out the proper divisor, we start with 1.0 and subtract off each
+            // preceeding test's likelihood of success.
+            //
+            unsigned const thenLikelihood = origCall->GetGDVCandidateInfo(checkIdx)->likelihood;
+            unsigned       baseLikelihood = 100;
+
+            for (uint8_t i = 0; i < checkIdx; i++)
+            {
+                baseLikelihood -= origCall->GetGDVCandidateInfo(i)->likelihood;
+            }
+
+            assert(baseLikelihood > 0);
```

Comment:
Since `baseLikelihood` is unsigned, are we just asserting that it's not zero here? If we're worried about underflow, then maybe we can flip the logic here by summing the likelihoods, and then checking that it's less than 100.

Reply:
The values should sum to less than 100, so let me change this bit.